intents/intent.ridl

```diff
@@ -101,19 +102,24 @@ struct IntentDataSessionAuthProof
 
 struct IntentDataSignMessage
   - network: string
-  - wallet:  string
+  - wallet: string
   - message: string
 
+struct IntentDataSignTypedData
+  - network: string
+  - wallet: string
+  - typedData: any
```

Comment:
Maybe we can add a `go.field.type` annotation to type it as `*ethcoder.TypedData` to avoid dynamic type conversion? (Though it would only work if that type implements a `MarshalJSON` that maps it back to the same JSON representation...)

Reply:
ok, done :)